src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,996 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Ak8963;
+using Iot.Units;
+using System;
+using System.Buffers.Binary;
+using System.Device;
+using System.Device.I2c;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Mpu9250
+{
+    /// <summary>
+    /// MPU9250 class. MPU9250 has an embedded gyroscope, accelerometer and temperature. It does offers a magnetometer thru an embedded AK8963.
+    /// </summary>
+    public class Mpu9250 : IDisposable
+    {
+        private const float Adc = 0x8000;
+        private const float Gravity = 9.807f;
+        private I2cDevice _i2cDevice;
+        private bool _autoDispose;
+        private Vector3 _accelerometerBias = new Vector3();
+        private Vector3 _gyroscopeBias = new Vector3();
+        private AccelerometerRange _accelerometerRange;
+        private GyroscopeRange _gyroscopeRange;
+        private AccelerometerBandwidth _accelerometerBandwidth;
+        private GyroscopeBandwidth _gyroscopeBandwidth;
+        private Ak8963.Ak8963 _ak8963;
+        private bool _wakeOnMotion;
+
+        /// <summary>
+        /// Default address for MPU9250
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x68;
+
+        /// <summary>
+        /// Second address for MPU9250
+        /// </summary>
+        public const byte SecondI2cAddress = 0x69;
+
+        #region Mgnetometer
+
+        /// <summary>
+        /// Get the magnetometer bias
+        /// </summary>
+        public Vector3 MagnometerBias => new Vector3(_ak8963.MagnometerBias.Y, _ak8963.MagnometerBias.X, -_ak8963.MagnometerBias.Z);
+
+        /// <summary>
+        /// Calibrate the magnetometer. Make sure your sensor is as far as possible of magnet
+        /// Calculate as well the magnetometer bias
+        /// </summary>
+        /// <returns>Returns the factory calibration data</returns>
+        public Vector3 CalibrateMagnetometer() => _wakeOnMotion ? Vector3.Zero : _ak8963.CalibrateMagnetometer();
+
+        /// <summary>
+        /// True if there is a data to read
+        /// </summary>
+        public bool DataReady => _wakeOnMotion ? false : _ak8963.DataReady;
```

Comment:
`!_wakeOnMotion && _ak8963.DataReady`?

Reply:
same comment for similar places below.

Also for DataReady as I've already mentioned before I'd recomment to change to something like WaitForData()